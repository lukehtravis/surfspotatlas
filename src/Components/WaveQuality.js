import React, {Component} from "react";
import ApolloClient from "apollo-boost";
import {graphql} from "react-apollo";
import gql from "graphql-tag";
import {WAVE_QUALITY} from "../utils/queries";
import PercentageCircle from "./PercentageCircle";
import WaveAttributeVote from "./WaveAttributeVote";
import { useAuth0 } from "../react-auth0-wrapper";

const WaveQuality = (props) => {

  let userId = "";
  const { isAuthenticated, user } = useAuth0()
  const waveQuality = props.attributeValue

  if (isAuthenticated) {
    userId = user.sub
  }

  return (
    <div>
      <PercentageCircle radius={50} percent={waveQuality}>
        <p>Epicness</p>
        <span>{waveQuality}%</span>
      </PercentageCircle>
      {isAuthenticated && (
        <WaveAttributeVote
          userId={userId}
          voteOnAttribute={props.voteOnAttribute}
          attributeValue={waveQuality}
          attributeName={props.attributeName}
        />)}
    </div>
  )
};

export default WaveQuality
