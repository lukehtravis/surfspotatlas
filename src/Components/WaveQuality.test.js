import React from "react";
import WaveQuality from "./WaveQuality"
import renderer from "react-test-renderer";
// import the auth0 hook
import { useAuth0 } from '../react-auth0-wrapper';

// intercept the useAuth0 function and mock it
jest.mock('../react-auth0-wrapper');
describe('Wave Quality', () => {
  test('Wave Quality Renders Properly when user logged in', () => {

    useAuth0.mockResolvedValue({isAuthenticated: true});
    const component = renderer.create(
      <WaveQuality />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  test('Wave Quality Renders Properly when user not logged in', () => {

    useAuth0.mockResolvedValue({isAuthenticated: false});
    const component = renderer.create(
      <WaveQuality />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
