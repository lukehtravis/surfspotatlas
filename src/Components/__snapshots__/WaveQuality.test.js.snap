// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wave Quality Wave Quality Renders Properly when user logged in 1`] = `
<div>
  <p
    className="MuiTypography-root WaveQuality-attributeHeader-1 MuiTypography-body1"
  >
    Wave Quality
  </p>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 16.99c-1.35 0-2.2.42-2.95.8-.65.33-1.18.6-2.05.6-.9 0-1.4-.25-2.05-.6-.75-.38-1.57-.8-2.95-.8s-2.2.42-2.95.8c-.65.33-1.17.6-2.05.6v1.95c1.35 0 2.2-.42 2.95-.8.65-.33 1.17-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.57.8 2.95.8s2.2-.42 2.95-.8c.65-.33 1.18-.6 2.05-.6.9 0 1.4.25 2.05.6.75.38 1.58.8 2.95.8v-1.95c-.9 0-1.4-.25-2.05-.6-.75-.38-1.6-.8-2.95-.8zm0-4.45c-1.35 0-2.2.43-2.95.8-.65.32-1.18.6-2.05.6-.9 0-1.4-.25-2.05-.6-.75-.38-1.57-.8-2.95-.8s-2.2.43-2.95.8c-.65.32-1.17.6-2.05.6v1.95c1.35 0 2.2-.43 2.95-.8.65-.35 1.15-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.57.8 2.95.8s2.2-.43 2.95-.8c.65-.35 1.15-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.58.8 2.95.8v-1.95c-.9 0-1.4-.25-2.05-.6-.75-.38-1.6-.8-2.95-.8zm2.95-8.08c-.75-.38-1.58-.8-2.95-.8s-2.2.42-2.95.8c-.65.32-1.18.6-2.05.6-.9 0-1.4-.25-2.05-.6-.75-.37-1.57-.8-2.95-.8s-2.2.42-2.95.8c-.65.33-1.17.6-2.05.6v1.93c1.35 0 2.2-.43 2.95-.8.65-.33 1.17-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.57.8 2.95.8s2.2-.43 2.95-.8c.65-.32 1.18-.6 2.05-.6.9 0 1.4.25 2.05.6.75.38 1.58.8 2.95.8V5.04c-.9 0-1.4-.25-2.05-.58zM17 8.09c-1.35 0-2.2.43-2.95.8-.65.35-1.15.6-2.05.6s-1.4-.25-2.05-.6c-.75-.38-1.57-.8-2.95-.8s-2.2.43-2.95.8c-.65.35-1.15.6-2.05.6v1.95c1.35 0 2.2-.43 2.95-.8.65-.32 1.18-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.57.8 2.95.8s2.2-.43 2.95-.8c.65-.32 1.18-.6 2.05-.6.9 0 1.4.25 2.05.6.75.38 1.58.8 2.95.8V9.49c-.9 0-1.4-.25-2.05-.6-.75-.38-1.6-.8-2.95-.8z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
    >
      <div
        className="makeStyles-root-144"
      >
        <div
          className="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-146 MuiLinearProgress-colorSecondary makeStyles-margin-145 MuiLinearProgress-determinate"
          role="progressbar"
        >
          <div
            className="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-147 MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
            style={Object {}}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Wave Quality Wave Quality Renders Properly when user not logged in 1`] = `
<div>
  <p
    className="MuiTypography-root WaveQuality-attributeHeader-1 MuiTypography-body1"
  >
    Wave Quality
  </p>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 16.99c-1.35 0-2.2.42-2.95.8-.65.33-1.18.6-2.05.6-.9 0-1.4-.25-2.05-.6-.75-.38-1.57-.8-2.95-.8s-2.2.42-2.95.8c-.65.33-1.17.6-2.05.6v1.95c1.35 0 2.2-.42 2.95-.8.65-.33 1.17-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.57.8 2.95.8s2.2-.42 2.95-.8c.65-.33 1.18-.6 2.05-.6.9 0 1.4.25 2.05.6.75.38 1.58.8 2.95.8v-1.95c-.9 0-1.4-.25-2.05-.6-.75-.38-1.6-.8-2.95-.8zm0-4.45c-1.35 0-2.2.43-2.95.8-.65.32-1.18.6-2.05.6-.9 0-1.4-.25-2.05-.6-.75-.38-1.57-.8-2.95-.8s-2.2.43-2.95.8c-.65.32-1.17.6-2.05.6v1.95c1.35 0 2.2-.43 2.95-.8.65-.35 1.15-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.57.8 2.95.8s2.2-.43 2.95-.8c.65-.35 1.15-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.58.8 2.95.8v-1.95c-.9 0-1.4-.25-2.05-.6-.75-.38-1.6-.8-2.95-.8zm2.95-8.08c-.75-.38-1.58-.8-2.95-.8s-2.2.42-2.95.8c-.65.32-1.18.6-2.05.6-.9 0-1.4-.25-2.05-.6-.75-.37-1.57-.8-2.95-.8s-2.2.42-2.95.8c-.65.33-1.17.6-2.05.6v1.93c1.35 0 2.2-.43 2.95-.8.65-.33 1.17-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.57.8 2.95.8s2.2-.43 2.95-.8c.65-.32 1.18-.6 2.05-.6.9 0 1.4.25 2.05.6.75.38 1.58.8 2.95.8V5.04c-.9 0-1.4-.25-2.05-.58zM17 8.09c-1.35 0-2.2.43-2.95.8-.65.35-1.15.6-2.05.6s-1.4-.25-2.05-.6c-.75-.38-1.57-.8-2.95-.8s-2.2.43-2.95.8c-.65.35-1.15.6-2.05.6v1.95c1.35 0 2.2-.43 2.95-.8.65-.32 1.18-.6 2.05-.6s1.4.25 2.05.6c.75.38 1.57.8 2.95.8s2.2-.43 2.95-.8c.65-.32 1.18-.6 2.05-.6.9 0 1.4.25 2.05.6.75.38 1.58.8 2.95.8V9.49c-.9 0-1.4-.25-2.05-.6-.75-.38-1.6-.8-2.95-.8z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
    >
      <div
        className="makeStyles-root-144"
      >
        <div
          className="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-146 MuiLinearProgress-colorSecondary makeStyles-margin-145 MuiLinearProgress-determinate"
          role="progressbar"
        >
          <div
            className="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-147 MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
            style={Object {}}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
